### PR TITLE
Setup KyberDao repository

### PR DESCRIPTION
**Pull request by manhlx3006**
Main contracts:
- KyberDao
- KyberStaking
- KyberFeeHandler
- EmergencyFeeHandler

### CODE REVIEW

File: solcOptimiserSettings.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  enabled: true,
+  runs: 430,
```
Comment (ilanDoron) [inline solcOptimiserSettings.js:3]: can set to 100000?

**Review by ilanDoron — COMMENTED** [review]
see comments.
when running coverage I get 65 failing tests. is this number of expected failures?

Also I See coverage almost full
is this the level we expect?

File: solcOptimiserSettings.js (hunk 1 above)
Comment (manhlx3006) [reply solcOptimiserSettings.js]: Updated to 1M runs, will have warning for KyberNetwork contract in Mock folder

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  include:
+    - env:
+        - TEST_PART: fullRegression
+
+install:
+  - yarn install
+
+before_script:
+  - yarn lint
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply .travis.yml:19]: Updated, it fails with js file

File: .solhintignore [hunk 3]
```diff
@@ -0,0 +1,10 @@
+## utils
+contracts/utils/*.sol
+contracts/mock/*.sol
+
+# production contracts
+contracts/IKyberFeeHandler.sol
+contracts/IKyberStaking.sol
+contracts/KyberDao.sol
+contracts/KyberFeeHandler.sol
+contracts/KyberStaking.sol
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply .solhintignore]: Updated

File: .solhintignore [hunk 4]
```diff
@@ -0,0 +1,10 @@
+## utils
+contracts/utils/*.sol
+contracts/mock/*.sol
+
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply .solhintignore:4]: Updated the ignore list

File: package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    "web3": "1.2.4",
+    "yargs": "^16.0.3"
+  },
+  "devDependencies": {
+    "@nomiclabs/buidler": "1.4.5",
+    "@nomiclabs/buidler-truffle5": "1.3.4",
+    "@nomiclabs/buidler-web3": "1.3.4",
+    "@openzeppelin/test-helpers": "0.5.6",
+    "chai": "4.2.0",
+    "chai-as-promised": "7.1.1",
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply package.json:37]: It is used in helper file, don't want to modify this file

File: package.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    "url": "https://github.com/KyberNetwork/kyber_dao_sc/issues"
+  },
+  "author": "Kyber SC Team",
+  "license": "MIT",
+  "dependencies": {
+    "dotenv": "8.2.0",
+    "mathjs": "4.4.2",
+    "solc": "0.6.6",
+    "solhint": "3.0.0-rc.8",
+    "truffle-assertions": "^0.9.2",
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply package.json]: yeah removed

File: package.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    "type": "git",
+    "url": "https://github.com/KyberNetwork/kyber_dao_sc.git"
+  },
+  "scripts": {
+    "test": "./test.sh",
+    "t": "./test.sh",
+    "compile": "./compile.sh",
+    "c": "./compile.sh",
+    "coverage": "./coverage.sh",
+    "lint": "yarn solhint 'contracts/*.sol' && yarn prettier -c './**/*.js'"
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply package.json]: ok updated, also fix path of js files

File: .prettierrc [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{
+  "overrides": [
+    {
+      "files": "*.sol",
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply .prettierrc]: removed

File: .solhint.json [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    "plugins": [],
+    "rules": {
+      "avoid-call-value": "error",
+      "avoid-low-level-calls": "off",
+      "avoid-sha3": "error",
+      "avoid-suicide": "error",
+      "avoid-throw": "error",
+      "avoid-tx-origin": "off",
+      "check-send-result": "error",
+      "compiler-version": ["off", "0.6.6"],
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply .solhint.json]: yeah only version 0.6.6

**Review by ilanDoron — COMMENTED** [review]
missing files:
daoFuzzTests.js.
stakingFuzzTests.js

I See many code changes in js files.
did u run preteffier?

**Comment by manhlx3006** [comment]
> missing files:
> daoFuzzTests.js.
> stakingFuzzTests.js
> 
> I See many code changes in js files.
> did u run preteffier?

Yes, alr ran prettier on all js files
will add daoFuzzTests and stakingFuzzTests

File: .prettierrc [hunk 10]
```diff
@@ -0,0 +1,16 @@
+{
+  "overrides": [
```
Comment (ducquangkstn) [inline .prettierrc:2]: Why do we remove prettier options for solidity?

**Review by ducquangkstn — APPROVED** [review]

**Comment by ducquangkstn** [comment]
We should move kybernetwork contracts file to contracts/mock/kyberNetwork

File: .prettierrc (hunk 10 above)
Comment (ilanDoron) [reply .prettierrc:2]: since we use only solhint
prettier creates some bad outputs.

**Review by ilanDoron — APPROVED** [review]

**Pull request merged**